docs/usage/configuration-options.md

```diff
@@ -1651,6 +1651,31 @@ For example to apply a special label for Major updates:
 }
 ```
 
+### replacementName
+
+Use this field to define the name of a replacement package.
+Should be used with `replacementVersion` (see example below).
+New package rules can be added by editing [`replacements.ts`](https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/replacements.ts)
+
+### replacementVersion
+
+Use this field to define the name of a replacement package.
+Should be used with `replacementVersion`.
```

Comment:
Must it, really? Can't one just declare the replacementVersion without the replacementName?

Reply:
I like having a fixed version. It's not like you could automatically take the latest version every time, e.g. if it's advanced to a new major release already

Reply:
> Must it, really? Can't one just declare the replacementVersion without the replacementName?

IMO, it should be stated somewhere that both are required to build a replacement PR. If you have only one package rule with just replacementVersion, it will be a no-op.

As far as I understand, replacementName and replacementVersion could be extracted by a datasource (none of them in this PR) or set/overriden by package rule. It seems package rules with preset is the way to go.
If a datasource could extract one or the other, a package rule may override the extracted values. Say the version will be the case most often. There is no matchReplacementName, so if multiple replacement exist, you would override the version of an unpredictable replacementName.

Still, you're right in that package rules config are additive. One can define the replacementName and another one the replacementVersion. I don't know if this has happened with other params and what's the recommended way to document it.

Reply:
I think it sounds safe for us to add a validation which makes sure they're in the same package rule

Reply:
Yeah, a change to a different dependency is most likely going to be a breaking change that will require manual intervention. Once the replacement PR is merged, a subsequent run of Renovate will upgrade to the latest version.